### DISCUSSION TITLE
Allow PR code suggestions on deleted lines

### DISCUSSION THREAD

**Discussion by jay** [community/community#37821] [Pull Requests]
I review PRs where sometimes in the name of refactoring or tidying up the author will remove a line that should not be removed. Maybe it should be changed though, and I would like to offer a suggestion other than removal. However GitHub will not allow this, and instead says _"Applying suggestions on deleted lines is not supported."_

I have done a search for similar discussions but only found #32114 which may be similar.

**Comment by jasonckeating** [comment]
In my use-case, there is a block of code that has a few required lines to secure it. If removed, without removing the entire block, it creates a security issue. I would like to be able to add a suggestion, or have automation drop a suggestion to keep the line or update as needed.

**Comment by daniram94** [comment]
i also found this useful for my daily work

**Comment by rikard-acst** [comment]
Yes, I would like to see the scenario where you would prefer to NOT be able to suggest on deleted lines?
